metadata.json

```diff
@@ -98,12 +98,13 @@
         "12.04",
         "14.04",
         "16.04",
-        "18.04"
+        "18.04",
+        "20.04"
       ]
     }
   ],
   "dependencies": [
     {"name":"trlinkin/nsswitch","version_requirement":">= 2.1.0 < 3.0.0"},
-    {"name":"puppetlabs/stdlib","version_requirement":">= 4.22.0 < 6.0.0"}
+    {"name":"puppetlabs/stdlib","version_requirement":">= 5.0.0 < 7.0.0"}
```

Comment:
Bumping the major to `<7` is no problem. What errors did you encounter with leaving the lower bound at `>= 4.22.0` ?

Reply:
The full output from the test suite is below; I'm not quite sure what makes it decide to try to pull in `stdlib v5.2.0`?

~~~
$ BEAKER_set="ubuntu-2004" BEAKER_PUPPET_COLLECTION=puppet6 bundle exec rake beaker
TEST_TIERS env variable not defined. Defaulting to run all tests.
/usr/bin/ruby2.5 -I/home/runejuhl/opt/puppet-module-pam/.bundle/ruby/2.5.0/gems/rspec-core-3.9.2/lib:/home/runejuhl/opt/puppet-module-pam/.bundle/ruby/2.5.0/gems/rspec-support-3.9.3/lib /home/runejuhl/opt/puppet-module-pam/.bundle/ruby/2.5.0/gems/rspec-core-3.9.2/exe/rspec spec/acceptance
/home/runejuhl/opt/puppet-module-pam/.bundle/ruby/2.5.0/gems/beaker-rspec-6.2.4/lib/beaker-rspec/helpers/serverspec.rb:43: warning: already initialized constant Module::VALID_OPTIONS_KEYS
/home/runejuhl/opt/puppet-module-pam/.bundle/ruby/2.5.0/gems/specinfra-2.82.16/lib/specinfra/configuration.rb:4: warning: previous definition of VALID_OPTIONS_KEYS was here
Hypervisor for ubuntu2004 is docker
Beaker::Hypervisor, found some docker boxes to create
get
/v1.16/version
{}

Provisioning docker
provisioning ubuntu2004
Creating image
Dockerfile is         FROM ubuntu:20.04
        ENV container docker
          RUN apt-get update
          RUN apt-get install -y openssh-server openssh-client curl ntpdate lsb-release apt-transport-https
        RUN mkdir -p /var/run/sshd
        RUN echo root:root | chpasswd
        RUN sed -ri 's/^#?PermitRootLogin .*/PermitRootLogin yes/' /etc/ssh/sshd_config
        RUN sed -ri 's/^#?PasswordAuthentication .*/PasswordAuthentication yes/' /etc/ssh/sshd_config
        RUN sed -ri 's/^#?UseDNS .*/UseDNS no/' /etc/ssh/sshd_config
RUN apt-get install -y -q net-tools wget locales
RUN locale-gen en_US.UTF-8
        EXPOSE 22
        CMD ["/sbin/init"]
Docker build buildargs: {}
post
/v1.16/build
{:rm=>true, :buildargs=>"{}"}
Dockerfile0000640000000000000000000000120513654311525013307 0ustar00wheelwheel00000000000000        FROM ubuntu:20.04
        ENV container docker
          RUN apt-get update
          RUN apt-get install -y openssh-server openssh-client curl ntpdate lsb-release apt-transport-https
        RUN mkdir -p /var/run/sshd
        RUN echo root:root | chpasswd
        RUN sed -ri 's/^#?PermitRootLogin .*/PermitRootLogin yes/' /etc/ssh/sshd_config
        RUN sed -ri 's/^#?PasswordAuthentication .*/PasswordAuthentication yes/' /etc/ssh/sshd_config
        RUN sed -ri 's/^#?UseDNS .*/UseDNS no/' /etc/ssh/sshd_config
RUN apt-get install -y -q net-tools wget locales
RUN locale-gen en_US.UTF-8
        EXPOSE 22
        CMD ["/sbin/init"]

get
/v1.16/containers/json
{}

Looking for an existing container with name pam-ubuntu2004
Existing container not found
Creating container from image c4b3e4f2607d
post
/v1.16/containers/create
{"name"=>"pam-ubuntu2004"}
{"Image":"c4b3e4f2607d","Hostname":"ubuntu2004","HostConfig":{"PortBindings":{"22/tcp":[{"HostPort":"7979","HostIp":"0.0.0.0"}]},"PublishAllPorts":true,"Privileged":true,"RestartPolicy":{"Name":"always"}}}
Starting container b37107b8a099b34065dafa44e3b92b13fecdcb2f2039a75373f5e76613c2ad43
post
/v1.16/containers/b37107b8a099b34065dafa44e3b92b13fecdcb2f2039a75373f5e76613c2ad43/start
{}
{}
get
/v1.16/containers/b37107b8a099b34065dafa44e3b92b13fecdcb2f2039a75373f5e76613c2ad43/json
{}

Using docker server at 0.0.0.0
get
/v1.16/containers/b37107b8a099b34065dafa44e3b92b13fecdcb2f2039a75373f5e76613c2ad43/json
{}

node available as  ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@0.0.0.0 -p 7979
get
/v1.16/containers/b37107b8a099b34065dafa44e3b92b13fecdcb2f2039a75373f5e76613c2ad43/json
{}


ubuntu2004 18:25:59$ cat /etc/resolv.conf
  Attempting ssh connection to 0.0.0.0, user: root, opts: {:password=>"root", :port=>"7979", :forward_agent=>false, :auth_methods=>["password", "publickey", "hostbased", "keyboard-interactive"]}
  Warning: Try 1 -- Host 0.0.0.0 unreachable: Errno::ECONNRESET - Connection reset by peer
  Warning: Trying again in 3 seconds
  Attempting ssh connection to 0.0.0.0, user: root, opts: {:password=>"root", :port=>"7979", :forward_agent=>false, :auth_methods=>["password", "publickey", "hostbased", "keyboard-interactive"], :logger=>#<Logger:0x000056313d7a2050 @level=4, @progname=nil, @default_formatter=#<Logger::Formatter:0x000056313d7a1f88 @datetime_format=nil>, @formatter=nil, @logdev=#<Logger::LogDevice:0x000056313d7a1b50 @shift_period_suffix=nil, @shift_size=nil, @shift_age=nil, @filename=nil, @dev=#<IO:<STDERR>>, @mon_owner=nil, @mon_count=0, @mon_mutex=#<Thread::Mutex:0x000056313d7a1a88>>>, :password_prompt=>#<Net::SSH::Prompt:0x000056313d7a17b8>, :user=>"root"}
  # Dynamic resolv.conf(5) file for glibc resolver(3) generated by resolvconf(8)
  #     DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN
  # 127.0.0.53 is the systemd-resolved stub resolver.
  # run "systemd-resolve --status" to see details about the actual nameservers.
  
  nameserver 10.4.6.1
  search lan wireless.local prod.sitad.dk petardo.lan

ubuntu2004 executed in 3.20 seconds

ubuntu2004 18:26:02$ echo '127.0.0.1	localhost localhost.localdomain
172.17.0.2	ubuntu2004.lan ubuntu2004
' >> /etc/hosts

ubuntu2004 executed in 0.01 seconds

ubuntu2004 18:26:02$ dpkg -s curl
  Package: curl
  Status: install ok installed
  Priority: optional
  Section: web
  Installed-Size: 401
  Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
  Architecture: amd64
  Multi-Arch: foreign
  Version: 7.68.0-1ubuntu2
  Depends: libc6 (>= 2.17), libcurl4 (= 7.68.0-1ubuntu2), zlib1g (>= 1:1.1.4)
  Description: command line tool for transferring data with URL syntax
   curl is a command line tool for transferring data with URL syntax, supporting
   DICT, FILE, FTP, FTPS, GOPHER, HTTP, HTTPS, IMAP, IMAPS, LDAP, LDAPS, POP3,
   POP3S, RTMP, RTSP, SCP, SFTP, SMTP, SMTPS, TELNET and TFTP.
   .
   curl supports SSL certificates, HTTP POST, HTTP PUT, FTP uploading, HTTP form
   based upload, proxies, cookies, user+password authentication (Basic, Digest,
   NTLM, Negotiate, kerberos...), file transfer resume, proxy tunneling and a
   busload of other useful tricks.
  Homepage: http://curl.haxx.se
  Original-Maintainer: Alessandro Ghedini <ghedo@debian.org>

ubuntu2004 executed in 0.06 seconds

ubuntu2004 18:26:02$ dpkg -s ntpdate
  Package: ntpdate
  Status: install ok installed
  Priority: optional
  Section: net
  Installed-Size: 174
  Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
  Architecture: amd64
  Source: ntp
  Version: 1:4.2.8p12+dfsg-3ubuntu4
  Depends: netbase, libc6 (>= 2.17), libssl1.1 (>= 1.1.0)
  Conffiles:
   /etc/default/ntpdate 39415ec9778476795fdbb832adc43b9b
   /etc/dhcp/dhclient-exit-hooks.d/ntpdate cb47fd9d3e21a204fb3ba4ca3fc8ab46
   /etc/logcheck/ignore.d.server/ntpdate 68d4df7cceb0e97bde87126c3a56b219
  Description: client for setting system time from NTP servers (deprecated)
   NTP, the Network Time Protocol, is used to keep computer clocks
   accurate by synchronizing them over the Internet or a local network,
   or by following an accurate hardware receiver that interprets GPS,
   DCF-77, NIST or similar time signals.
   .
   ntpdate is deprecated. Please use sntp instead for manual or scripted
   NTP queries/syncs.
  Homepage: http://support.ntp.org/
  Original-Maintainer: Debian NTP Team <ntp@packages.debian.org>

ubuntu2004 executed in 0.02 seconds
setting local environment on ubuntu2004

ubuntu2004 18:26:02$ mktemp -dt .XXXXXX
  /tmp/.BMyaRy

ubuntu2004 executed in 0.05 seconds

ubuntu2004 18:26:02$ echo 'PermitUserEnvironment yes' | cat - /etc/ssh/sshd_config > /tmp/.BMyaRy/sshd_config.permit

ubuntu2004 executed in 0.01 seconds

ubuntu2004 18:26:02$ mv /tmp/.BMyaRy/sshd_config.permit /etc/ssh/sshd_config

ubuntu2004 executed in 0.01 seconds

ubuntu2004 18:26:02$ echo '' >/etc/environment

ubuntu2004 executed in 0.05 seconds

ubuntu2004 18:26:02$ service ssh restart

ubuntu2004 executed in 0.06 seconds

ubuntu2004 18:26:02$ mkdir -p ~/.ssh

ubuntu2004 executed in 0.01 seconds

ubuntu2004 18:26:03$ chmod 0600 ~/.ssh

ubuntu2004 executed in 0.01 seconds

ubuntu2004 18:26:03$ touch ~/.ssh/environment

ubuntu2004 executed in 0.01 seconds

ubuntu2004 18:26:03$ grep ^PATH=.*\$PATH ~/.ssh/environment

ubuntu2004 executed in 0.01 seconds
Exited: 1

ubuntu2004 18:26:03$ grep ^PATH= ~/.ssh/environment

ubuntu2004 executed in 0.01 seconds
Exited: 1

ubuntu2004 18:26:03$ echo "PATH=$PATH" >> ~/.ssh/environment

ubuntu2004 executed in 0.01 seconds
will not mirror environment to /etc/profile.d on non-sles platform host
ssh connection to ubuntu2004 has been terminated

ubuntu2004 18:26:03$ cat ~/.ssh/environment
  Attempting ssh connection to 0.0.0.0, user: root, opts: {:password=>"root", :port=>"7979", :forward_agent=>false, :auth_methods=>["password", "publickey", "hostbased", "keyboard-interactive"], :logger=>#<Logger:0x000056313d7a2050 @level=4, @progname=nil, @default_formatter=#<Logger::Formatter:0x000056313d7a1f88 @datetime_format=nil>, @formatter=nil, @logdev=#<Logger::LogDevice:0x000056313d7a1b50 @shift_period_suffix=nil, @shift_size=nil, @shift_age=nil, @filename=nil, @dev=#<IO:<STDERR>>, @mon_owner=nil, @mon_count=0, @mon_mutex=#<Thread::Mutex:0x000056313d7a1a88>>>, :password_prompt=>#<Net::SSH::Prompt:0x000056313d7a17b8>, :user=>"root"}
  PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games

ubuntu2004 executed in 0.16 seconds
Disabling updates.puppetlabs.com by modifying hosts file to resolve updates to 127.0.0.1 on ubuntu2004

ubuntu2004 18:26:03$ echo '127.0.0.1	updates.puppetlabs.com
' >> /etc/hosts

ubuntu2004 executed in 0.01 seconds

ubuntu2004 18:26:03$ wget -O /tmp/puppet.deb http://apt.puppet.com/puppet6-release-focal.deb
  --2020-05-05 16:26:03--  http://apt.puppet.com/puppet6-release-focal.deb
  Resolving apt.puppet.com (apt.puppet.com)...   52.85.241.13, 52.85.241.20, 52.85.241.39, ...
  Connecting to apt.puppet.com (apt.puppet.com)|52.85.241.13|:80...   connected.
  HTTP request sent, awaiting response...   200 OK
  Length: 11732 (11K) [application/x-debian-package]
  Saving to: '/tmp/puppet.deb'
  
       0K .......... .                                            100% 3.26M=0.003s
  2020-05-05 16:26:03 (3.26 MB/s) - '/tmp/puppet.deb' saved [11732/11732]

ubuntu2004 executed in 0.29 seconds

ubuntu2004 18:26:03$ dpkg -i --force-all /tmp/puppet.deb
  Selecting previously unselected package puppet6-release.
  (Reading database ... 12920 files and directories currently installed.)
  Preparing to unpack /tmp/puppet.deb ...
  Unpacking puppet6-release (6.0.0-9focal) ...
  Setting up puppet6-release (6.0.0-9focal) ...
  
  Configuration file '/etc/apt/sources.list.d/puppet6.list', does not exist on system.
  Installing new config file as you requested.
  
  Configuration file '/etc/apt/trusted.gpg.d/puppet6-keyring.gpg', does not exist on system.
  Installing new config file as you requested.

ubuntu2004 executed in 0.25 seconds

ubuntu2004 18:26:03$ apt-get update
  Hit:1 http://archive.ubuntu.com/ubuntu focal InRelease
  Hit:2 http://security.ubuntu.com/ubuntu focal-security InRelease
  Hit:3 http://archive.ubuntu.com/ubuntu focal-updates InRelease
  Hit:4 http://archive.ubuntu.com/ubuntu focal-backports InRelease
  Get:5 http://apt.puppetlabs.com focal InRelease [28.3 kB]
  Get:6 http://apt.puppetlabs.com focal/puppet6 amd64 Packages [1574 B]
  Get:7 http://apt.puppetlabs.com focal/puppet6 all Packages [739 B]
  Fetched 30.6 kB in 1s (20.9 kB/s)
  Reading package lists...

ubuntu2004 executed in 2.17 seconds

ubuntu2004 18:26:06$ echo "/opt/puppetlabs/bin"
  /opt/puppetlabs/bin

ubuntu2004 executed in 0.01 seconds

ubuntu2004 18:26:06$ echo "/opt/puppet-git-repos/hiera/bin"
  /opt/puppet-git-repos/hiera/bin

ubuntu2004 executed in 0.05 seconds

ubuntu2004 18:26:06$ echo "/opt/puppetlabs/puppet/bin"
  /opt/puppetlabs/puppet/bin

ubuntu2004 executed in 0.01 seconds

ubuntu2004 18:26:06$ grep ^PATH=.*\/opt\/puppetlabs\/bin:\/opt\/puppet\-git\-repos\/hiera\/bin:\/opt\/puppetlabs\/puppet\/bin ~/.ssh/environment

ubuntu2004 executed in 0.01 seconds
Exited: 1

ubuntu2004 18:26:06$ grep ^PATH= ~/.ssh/environment
  PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games

ubuntu2004 executed in 0.05 seconds

ubuntu2004 18:26:06$ sed -i -e "s/^PATH=/PATH=\/opt\/puppetlabs\/bin:\/opt\/puppet\-git\-repos\/hiera\/bin:\/opt\/puppetlabs\/puppet\/bin:/" ~/.ssh/environment

ubuntu2004 executed in 0.01 seconds
will not mirror environment to /etc/profile.d on non-sles platform host

ubuntu2004 18:26:06$ grep ^PATH=.*PATH ~/.ssh/environment

ubuntu2004 executed in 0.01 seconds
Exited: 1

ubuntu2004 18:26:06$ grep ^PATH= ~/.ssh/environment
  PATH=/opt/puppetlabs/bin:/opt/puppet-git-repos/hiera/bin:/opt/puppetlabs/puppet/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games

ubuntu2004 executed in 0.05 seconds

ubuntu2004 18:26:06$ sed -i -e "s/^PATH=/PATH=PATH:/" ~/.ssh/environment

ubuntu2004 executed in 0.01 seconds
will not mirror environment to /etc/profile.d on non-sles platform host

ubuntu2004 18:26:06$ apt-get update
  Hit:1 http://archive.ubuntu.com/ubuntu focal InRelease
  Hit:2 http://security.ubuntu.com/ubuntu focal-security InRelease
  Hit:3 http://archive.ubuntu.com/ubuntu focal-updates InRelease
  Hit:4 http://archive.ubuntu.com/ubuntu focal-backports InRelease
  Hit:5 http://apt.puppetlabs.com focal InRelease
  Reading package lists...

ubuntu2004 executed in 1.53 seconds

ubuntu2004 18:26:07$ apt-get install --force-yes  -y puppet-agent
  Reading package lists...
  Building dependency tree...  
  Reading state information...
  The following NEW packages will be installed:
    puppet-agent
  0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
  Need to get 22.5 MB of archives.
  After this operation, 136 MB of additional disk space will be used.
  Get:1 http://apt.puppetlabs.com focal/puppet6 amd64 puppet-agent amd64 6.15.0-1focal [22.5 MB]
  debconf: delaying package configuration, since apt-utils is not installed
  Fetched 22.5 MB in 4s (6180 kB/s)
  Selecting previously unselected package puppet-agent.
  (Reading database ...   (Reading database ... 5%(Reading database ... 10%(Reading database ... 15%(Reading database ... 20%(Reading database ... 25%  (Reading database ... 30%(Reading database ... 35%(Reading database ... 40%(Reading database ... 45%(Reading database ... 50%(Reading database ... 55%(Reading database ... 60%  (Reading database ... 65%  (Reading database ... 70%  (Reading database ... 75%  (Reading database ... 80%  (Reading database ... 85%  (Reading database ... 90%  (Reading database ... 95%  (Reading database ... 100%(Reading database ... 12925 files and directories currently installed.)
  Preparing to unpack .../puppet-agent_6.15.0-1focal_amd64.deb ...
  Unpacking puppet-agent (6.15.0-1focal) ...
  Setting up puppet-agent (6.15.0-1focal) ...
  Created symlink /etc/systemd/system/multi-user.target.wants/puppet.service → /lib/systemd/system/puppet.service.
  Created symlink /etc/systemd/system/multi-user.target.wants/pxp-agent.service → /lib/systemd/system/pxp-agent.service.
  Removed /etc/systemd/system/multi-user.target.wants/pxp-agent.service.
  Processing triggers for libc-bin (2.31-0ubuntu9) ...
  W: --force-yes is deprecated, use one of the options starting with --allow instead.

ubuntu2004 executed in 8.70 seconds

ubuntu2004 18:26:16$ echo "/opt/puppetlabs/bin"
  /opt/puppetlabs/bin

ubuntu2004 executed in 0.01 seconds

ubuntu2004 18:26:16$ echo "/opt/puppet-git-repos/hiera/bin"
  /opt/puppet-git-repos/hiera/bin

ubuntu2004 executed in 0.01 seconds

ubuntu2004 18:26:16$ echo "/opt/puppetlabs/puppet/bin"
  /opt/puppetlabs/puppet/bin

ubuntu2004 executed in 0.01 seconds

ubuntu2004 18:26:16$ grep ^PATH=.*\/opt\/puppetlabs\/bin:\/opt\/puppet\-git\-repos\/hiera\/bin:\/opt\/puppetlabs\/puppet\/bin ~/.ssh/environment
  PATH=PATH:/opt/puppetlabs/bin:/opt/puppet-git-repos/hiera/bin:/opt/puppetlabs/puppet/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games

ubuntu2004 executed in 0.01 seconds

ubuntu2004 18:26:16$ grep ^PATH=.*PATH ~/.ssh/environment
  PATH=PATH:/opt/puppetlabs/bin:/opt/puppet-git-repos/hiera/bin:/opt/puppetlabs/puppet/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games

ubuntu2004 executed in 0.01 seconds

ubuntu2004 18:26:18$ puppet module install trlinkin-nsswitch -v 2.2.0
  Notice: Preparing to install into /etc/puppetlabs/code/environments/production/modules ...
  Notice: Downloading from https://forgeapi.puppet.com ...
  Notice: Installing -- do not interrupt ...
  /etc/puppetlabs/code/environments/production/modules
  └─┬ trlinkin-nsswitch (v2.2.0)
    └── puppetlabs-stdlib (v6.3.0)

ubuntu2004 executed in 5.16 seconds

ubuntu2004 18:26:23$ puppet module install puppetlabs-stdlib -v 5.2.0
  Notice: Preparing to install into /etc/puppetlabs/code/environments/production/modules ...
  Error: Could not install module 'puppetlabs-stdlib' (v5.2.0)
    Module 'puppetlabs-stdlib' (v6.3.0) is already installed
      Use `puppet module upgrade` to install a different version
      Use `puppet module install --force` to re-install only this module

ubuntu2004 executed in 1.14 seconds
Exited: 1

An error occurred while loading ./spec/acceptance/00_pam_spec.rb.
Failure/Error: install_module_dependencies
Beaker::Host::CommandFailure:
  Host 'ubuntu2004' exited with 1 running:
   puppet module install puppetlabs-stdlib -v 5.2.0
  Last 10 lines of output were:
  	Notice: Preparing to install into /etc/puppetlabs/code/environments/production/modules ...
  	Error: Could not install module 'puppetlabs-stdlib' (v5.2.0)
  	  Module 'puppetlabs-stdlib' (v6.3.0) is already installed
  	    Use `puppet module upgrade` to install a different version
  	    Use `puppet module install --force` to re-install only this module
  
# ./.bundle/ruby/2.5.0/gems/beaker-4.21.0/lib/beaker/host.rb:395:in `exec'
# ./.bundle/ruby/2.5.0/gems/beaker-4.21.0/lib/beaker/dsl/helpers/host_helpers.rb:83:in `block in on'
# ./.bundle/ruby/2.5.0/gems/beaker-4.21.0/lib/beaker/shared/host_manager.rb:130:in `run_block_on'
# ./.bundle/ruby/2.5.0/gems/beaker-4.21.0/lib/beaker/dsl/patterns.rb:37:in `block_on'
# ./.bundle/ruby/2.5.0/gems/beaker-4.21.0/lib/beaker/dsl/helpers/host_helpers.rb:63:in `on'
# ./.bundle/ruby/2.5.0/gems/beaker-puppet-1.18.15/lib/beaker-puppet/install_utils/module_utils.rb:84:in `block in install_puppet_module_via_pmt_on'
# ./.bundle/ruby/2.5.0/gems/beaker-4.21.0/lib/beaker/shared/host_manager.rb:130:in `run_block_on'
# ./.bundle/ruby/2.5.0/gems/beaker-4.21.0/lib/beaker/dsl/patterns.rb:37:in `block_on'
# ./.bundle/ruby/2.5.0/gems/beaker-puppet-1.18.15/lib/beaker-puppet/install_utils/module_utils.rb:62:in `install_puppet_module_via_pmt_on'
# ./.bundle/ruby/2.5.0/gems/beaker-module_install_helper-0.1.7/lib/beaker/module_install_helper.rb:40:in `block (2 levels) in install_module_dependencies_on'
# ./.bundle/ruby/2.5.0/gems/beaker-module_install_helper-0.1.7/lib/beaker/module_install_helper.rb:38:in `each'
# ./.bundle/ruby/2.5.0/gems/beaker-module_install_helper-0.1.7/lib/beaker/module_install_helper.rb:38:in `block in install_module_dependencies_on'
# ./.bundle/ruby/2.5.0/gems/beaker-module_install_helper-0.1.7/lib/beaker/module_install_helper.rb:37:in `each'
# ./.bundle/ruby/2.5.0/gems/beaker-module_install_helper-0.1.7/lib/beaker/module_install_helper.rb:37:in `install_module_dependencies_on'
# ./.bundle/ruby/2.5.0/gems/beaker-module_install_helper-0.1.7/lib/beaker/module_install_helper.rb:25:in `install_module_dependencies'
# ./spec/spec_helper_acceptance.rb:7:in `<top (required)>'
# ./spec/acceptance/00_pam_spec.rb:1:in `require'
# ./spec/acceptance/00_pam_spec.rb:1:in `<top (required)>'
No examples found.


Finished in 0.00004 seconds (files took 1 minute 31.67 seconds to load)
0 examples, 0 failures, 1 error occurred outside of examples

zlib(finalizer): the stream was freed prematurely.
/usr/bin/ruby2.5 -I/home/runejuhl/opt/puppet-module-pam/.bundle/ruby/2.5.0/gems/rspec-core-3.9.2/lib:/home/runejuhl/opt/puppet-module-pam/.bundle/ruby/2.5.0/gems/rspec-support-3.9.3/lib /home/runejuhl/opt/puppet-module-pam/.bundle/ruby/2.5.0/gems/rspec-core-3.9.2/exe/rspec spec/acceptance failed
~~~

Reply:
This is because the dependency installs a newer stdlib. It's lower bound for stdlib is 4.25.0.

Reply:
Please change to `">= 4.25.0 < 7.0.0"`